src/main/java/chess/controller/SpringChessApiController.java

```diff
@@ -0,0 +1,72 @@
+package chess.controller;
+
+import chess.dto.web.BoardDto;
+import chess.dto.web.GameStatusDto;
+import chess.dto.web.MovementDto;
+import chess.dto.web.PointsDto;
+import chess.dto.web.RoomDto;
+import chess.dto.web.UsersInRoomDto;
+import chess.service.ChessService;
+import org.eclipse.jetty.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/rooms")
+public class SpringChessApiController {
+
+    private final ChessService chessService;
+
+    public SpringChessApiController(ChessService chessService) {
+        this.chessService = chessService;
+    }
+
+    @PostMapping
+    private ResponseEntity<RoomDto> createRoom(@RequestBody RoomDto roomDto) {
+        return ResponseEntity.status(HttpStatus.CREATED_201).body(chessService.create(roomDto));
+    }
+
+    @PutMapping
+    private ResponseEntity<Object> closeRoom(@RequestBody RoomDto roomDto) {
+        chessService.close(roomDto.getId());
+        return ResponseEntity.status(HttpStatus.CREATED_201).build();
+    }
```

Comment:
1. 응답의 바디가 없을 땐 `ResponseEntity<Void>`를 리턴하면 됩니다.
2. `PUT rooms/{id}/close`에서 `PUT rooms`가 된 건 **자원에 대한 행위는 HTTP Method(GET, POST, PUT, DELETE)로 표현한다**는 원칙 때문일까요? 하지만 HTTP 메서드는 개수가 정말 적어서 우리가 원하는 행위를 제대로 표현하지 못하는 경우가 많죠. `PUT rooms`만 보면 방을 닫는 엔드포인트라는 걸 전혀 예상하지 못할 것 같은데 루트는 어떻게 생각하시나요?
3. 또한 응답으로 `201 CREATED`가 적절한지도 고민해 보아요. 아래 엔드포인트들도 마찬가지입니다.

Reply:
1. 👍👍👍👍
2. REST의 의미를 만족하기 위해서는 그래도 저는 명사인게 좋지않을까 싶어요. 그리고 이번에 처음으로 REST라는 개념을 알게되어서 좀 더 연습해보기위해 극단적인 연습도 해보면 좋을거같다는 생각도 드네요 😊 지금 같은 경우는 확실히 PUT만으로는 방을 닫는 행위라고 이해하기 힘들거같아요. 그래서 리소스를 좀 더 구체화 시켜보는 건 어떨까 싶어요. `PUT rooms/{id}/status` `body {status: "closed"}` 이런식으로요!
3. 위에 CREATED가 왜 만들어졌는지 학습하면서 PUT에는 맞지않는 응답코드라는 생각이들었어요!!

Reply:
DELETE 메서드는 리소스를 삭제할 때 사용되는데 방을 닫는 것이 리소스를 삭제하는 것과 같다고 보기는 어려울 것 같아요. `PUT rooms/{id}/status body {status: "closed"}` 방식이 괜찮아 보이네요.

Reply:
여기서 한가지 의문점이 더 생기는데요. 리소스라는게 누구의 관점에서 사용되는 용어인지 잘 모르곘어요. 현재 db입장에서는 `closed` 속성을 `true`로 바꾸는 것이지만, 사실 삭제를 의미하는 '논리 삭제'를 구현해뒀어요. 클라이언트에서는 db가 어떻게 처리하는지 신경쓰지 않고 단순히 방을 삭제하는 것으로 인식한다면 `DELETE` 메서드가 맞다고 생각이 드네요. 리소스를 클라이언트 입장에서 봐도 괜찮을까요?

Reply:
> ### 자원과 자원 식별자
> REST에서 정보의 핵심 추상화는 자원(resource)이다.
> - 이름을 지정할 수 있는 세상의 모든 것은 다 자원이 될 수 있음.
> - 시간이 지남에 따라 값이 바뀌는 것이 있고, 그렇지 않은 것이 있다.
> - 자원은 "특정 시점의 매핑에 해당하는 엔티티"가 아니라 "엔티티 집합에 대한 개념적 매핑"이다.
>   - 자원의 타입이나 구현에 영향을 받지 않는 보편성을 제공한다.
>   - 요청의 특성을 기반으로 콘텐츠 협상을 수행 할 수 있도록 표현에 대한 참조의 지연 바인딩을 허용한다.
>   - 개념의 단수 표현이 아니라 집합에 대한 표현이므로 표현이 바뀌어도 링크를 바꾸지 않아도 된다.
>
> [(요약) Architectural Styles and the Design of Network-based Software Architectures by Roy Thomas Fielding, 2000 ](https://johngrib.github.io/wiki/REST-paper-summary/#5211-resources-and-resource-identifiers)

상당히 개념적이고 어렵죠 😅 저도 잘 아는 것이 아니라서 섣불리 단정지어 설명드릴 수는 없지만, 자원이라는 용어가 서버 혹은 클라이언트 중 특정 사이드의 관점에서 사용되는 것 같지는 않네요.

> ### Rule: DELETE must be used to remove a resource from its parent
> A client uses DELETE to request that a resource be completely removed from its parent, which is often a collection or store. Once a DELETE request has been processed for a given resource, the resource can no longer be found by clients. Therefore, any future attempt to retrieve the resource’s state representation, using either GET or HEAD, must result in a 404 ("Not Found") status returned by the API.
>
> The DELETE method has very specific semantics in HTTP, which must not be overloaded or stretched by a REST API’s design. Specifically, an API should not distort the intended meaning of DELETE by mapping it to a lesser action that leaves the resource, and its URI, available to clients. For example, if an API wishes to provide a "soft" delete or some other state-changing interaction, it should employ a special controller resource and direct its clients to use POST instead of DELETE to interact.
>
> [REST API Design Rulebook](https://www.oreilly.com/library/view/rest-api-design/9781449317904/)

DELETE 메서드에 관해서라면 위의 인용이 참고할 만할 것 같아요.

Reply:
`DELETE`의 경우에는 해당 리소스에 다시 접근할 수 있는 가능성이 절대 없을 때만 써야되나 보군요 😯
REST를 완벽하게 이해하기에는 아직 공부해야되는 부분이 많을 것 같아요 ㅠㅠ
일단 위의 `PUT` 요청으로 수정해보겠습니다!

Reply:
> REST의 의미를 만족하기 위해서는 그래도 저는 명사인게 좋지않을까 싶어요. 그리고 이번에 처음으로 REST라는 개념을 알게되어서 좀 더 연습해보기위해 극단적인 연습도 해보면 좋을거같다는 생각도 드네요 😊

여기에 대해 추가 코멘트를 남기려다 말았는데 아무래도 남기는 게 좋을 것 같아서 추가합니다.

우선 REST API 디자인 원칙 == URI 설계 원칙이 아닙니다. 사실 어떤 단일한 원칙이 존재하는 것도 아니에요. 저는 URI를 리소스 중심으로 설계하라는 것, 그리고 HTTP 메서드로 이미 표현되고 있는 행위를 중복해서 드러내지 말라는 것 등을 참고하시라고 남긴 것이었어요. 제가 드린 [첫 코멘트](https://github.com/woowacourse/jwp-chess/pull/262#discussion_r616609535)의 코드 블록을 보시면

```java
@GetMapping("{id}/getGameStatus")
```

위와 같이 메서드로 유추할 수 있는 `get`이 URI에서도 중복 사용되고 있었죠?

따라서 "URI에 명사만 들어가야 한다!"는 것은 오해이며, 어떤 원칙들을 따르는 데 집중하는 것보단 그 원칙이 탄생한 배경, 그리고 목적 등에 관해 생각해 보셨으면 좋겠어요!

Reply:
추가 코멘트 고마워요 코니! 
미션 완료한 후에 추가적으로 REST에대해 학습해봤어요! 학습하면서 코니가 우려한 `REST API 디자인 원칙 == URI 설계 원칙` 이 아니였다는 점에서 놀랐어요. REST라는 디자인 제약은 꼭 HTTP만을 위한 것도 아니였으며, 리소스 중심으로 URI를 설계하면서 일관성 있는 인터페이스를 만드는 것이 주목적이라고 이해했어요. 학습하면서 제가 REST를 완전히 오해하고 있었다는 사실을 알게되었는데, 코니가 한번더 언급해주면서 이제는 확실하게 오해하지 않을 것 같아요. 감사합니다~ 🙏

<https://junroot.github.io/programming/REST/>